### DISCUSSION TITLE
Create microcomponents for each language

### DISCUSSION THREAD

**Discussion by luilliarcec** [Laravel-Lang/common#49] [Q&A] [answered]
If microcomponents are created for each language, it would help to only install the required language, by the developer, in addition these microcomponents can be included in turn in "lang" for full use of all existing languages ​​if required. I leave this idea that came to me at your discretion, first of all thanks for the work done.

**Comment by caouecs** [comment]
Hello,
thank you for your idea, but microwhat ? Can you explain please ?

**Reply by luilliarcec** [reply]
Create a package, for example, for the spanish language, another for english, and so on. 

This way you could keep each language independently and then incorporate them as dependencies to Lang package.
In addition to giving the possibility of installing a single language or only those required to the developer, without losing the updates of those languages.

**Comment by caouecs** [comment, marked as answer]
good luck to manage ;)

**Comment by andrey-helldar** [comment]
@luilliarcec, on the one hand, the good idea is that it will simplify the work of maintainers - everyone will follow their own language.

On the other hand, this is very, very bad. Here you say "install" separately. To install, you will need to run a command, for example `composer require laravel-lang/lang-{lang}`. Now imagine that the language needs to be added in production via the web interface (for example, [tanmuhittin/laravel-google-translate](https://github.com/tanmuhittin/laravel-google-translate))...

Now, in order to install the desired language, you can use the localization managers - [Lang Publisher](https://github.com/andrey-helldar/laravel-lang-publisher), [LaravelLang](https://github.com/ARCANEDEV/LaravelLang) and [Laravel-lang](https://github.com/overtrue/laravel-lang). What prevents you from using one of them?

**Reply by andrey-helldar** [reply]
I remembered another problem with dividing the project into micro-components - in each of the components we will need to inject a script that compares the difference with the main files (todo.php) and keep him up to date. This will greatly complicate the task.
